quantum/rgblight_post_config.h

```diff
@@ -3,3 +3,37 @@
 // it is considered that RGBLIGHT_SPLIT is defined implicitly.
 #    define RGBLIGHT_SPLIT
 #endif
+
+#ifdef RGBLIGHT_ANIMATIONS
+// for backward compatibility
+#    define RGBLIGHT_EFFECT_BREATHING
+#    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#    define RGBLIGHT_EFFECT_SNAKE
+#    define RGBLIGHT_EFFECT_KNIGHT
+#    define RGBLIGHT_EFFECT_CHRISTMAS
+#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#    define RGBLIGHT_EFFECT_RGB_TEST
+#    define RGBLIGHT_EFFECT_ALTERNATING
```

Comment:
Could these two be moved to a different define, something like:

```
#ifdef RGBLIGHT_ANIMATIONS_TEST
#    define RGBLIGHT_EFFECT_RGB_TEST
#    define RGBLIGHT_EFFECT_ALTERNATING
#endif

Reply:
I found #7680 and #7669. However, I want to focus on refactoring rgblight_reconfig.h in this PR.

Reply:
Okay